easybuild/framework/easyconfig/tweak.py

```diff
@@ -767,7 +773,10 @@ def map_toolchain_hierarchies(source_toolchain, target_toolchain, modtool):
     target_tc_hierarchy = get_toolchain_hierarchy(target_toolchain, incl_capabilities=True)
 
     for toolchain_spec in source_tc_hierarchy:
-        tc_mapping[toolchain_spec['name']] = match_minimum_tc_specs(toolchain_spec, target_tc_hierarchy)
+        minimal_matching_toolchain = match_minimum_tc_specs(toolchain_spec, target_tc_hierarchy)
+        if not minimal_matching_toolchain:
+            return None
+        tc_mapping[toolchain_spec['name']] = minimal_matching_toolchain
```

Comment:
Same here, please use positive logic + `if`/`else`, and maybe also add a debug log message:

```python
if minimal_matching_toolchain:
    tc_mapping[toolchain_spec['name']] = minimal_matching_toolchain
else:
    _log.debug("No minimal matching toolchain found for %s using %s", toolchain_spec, target_tc_hierarchy)
    return None
```

I'm also not a big fan of the inline `return` statement. Can we avoid this?